tools/generator/Java.Interop.Tools.Generator.ObjectModel/Method.cs

```diff
@@ -64,7 +64,8 @@ public string AutoDetectEnumifiedOverrideReturn (AncestorDescendantCache cache)
 
 		public bool CanAdd {
 			get {
-				return Name.Length > 3 && Name.StartsWith ("Add") && Name.EndsWith ("Listener") && Parameters.Count == 1 && IsVoid &&
+				return Name.Length > 3 && Name.StartsWith ("Add") && Name.EndsWith ("Listener") &&  IsVoid &&
+					(Parameters.Count == 1 || (Parameters.Count == 2 && Parameters[1].Name == "handler")) &&
```

Comment:
Instead of checking the parameter name, this should check the parameter type, that it *is* `Android.OS.Handler`.

Reply:
Thanks for the great suggestion!